In [1]:
import pandas as pd
import numpy as np
import validators


In [2]:
tabsv = "data/off_dataset.tsv"

df_off = pd.read_csv(tabsv, sep="\t", dtype="unicode", low_memory=False)


In [72]:
print(f"""There is {len(df_off)} products in the dataset and\
 only {len(df_off[df_off['carbon-footprint_100g'].notna()])} contain their carbon footprint
This is not enough to draw an average
 """)

There is 320772 products in the dataset and only 268 contain their carbon footprint
This is not enough to draw an average
 


In [91]:
len(df_off[df_off["stores"].notna()])

51722

Among the 320722 products, only 51722 have non NaN "store" field. Building an application using the stores data won't be efficient in itself, but if the user of the potential application is invited to fill the store's field of a product, this could lead to interesting functionnalities

In [36]:
# Filters

def get_col_by_ending(df: pd.DataFrame, ending: str) -> list:
    """takes an ending and a df as parameters, returns all cols ending with parameter"""
    
    col_list = []

    for col in df.columns:
        if col.endswith(ending):
            col_list.append(col)

    return col_list



def filter_series(series: pd.Series):
    """
    set of filter : if not url -> False, if not proper URL
    -> False, no name -> False, no nutriscore -> False
    """
    if pd.isna(series["url"]):
        return False
    if not validators.url(series["url"]):
        return False
    if pd.isna(series["product_name"]):
        return False
    if pd.isna(series["nutrition_grade_fr"]):
        return False

def filter_creator(creator: str) -> bool:
    pass


def ecolo_rate(series: pd.Series) -> dict:
    """
    returns if a products contains palm oil and/or if its labelled as bio
    0 -> unknown , False -> No, True -> Yes
    """
    bad_stuff = {
        "palm_oil": 0,
        "bio": 0,
    }
    pass


def aberation_cleaner(series: pd.Series, hundred_g_list: list):
    """if a non na col of a 100g is > 100 or < 0, returns false""""

    for col in hundred_g_list:
        if pd.notna(series[col]):
            if float(series[col]) > 100 or float(series[col]) < 0:
                return False


def copy_cleaner(df: pd.DataFrame, subset: str):
    pass


In [39]:
hundred_g_list = get_col_by_ending(df_off, "100g")

In [82]:
print("products", len(df_off))
print("unique products", len(df_off["code"].unique()))

products 320772
unique products 320750


In [84]:
cc1 = "Coca Cola 50Cl"
cc2 = "Coca Cola 500ml"
cc3 = "Coca-Cola 50 Cl"
cc4 = "CocaCola 50Cl"
cc5 = "Coca-Cola 500ml"
cc6 = "Coca Cola .50L"
cc7 = "Coca Cola0,5L"
cc8 = "Coca Cola.5L"
cc9 = " cocacola 50cl"
cc10 = "coke cola 50cl" #  This exemple will not work

ccl = [cc1, cc2, cc3, cc4, cc5, cc6, cc7, cc8, cc9, cc10]


In [85]:
def detect_name_copy(analyzed_name: str, original_name: str) -> bool:
    pass

def parse_data(product_name: str):
    pass

def ean_thirteen_filter(df: pd.DataFrame):
    df.drop_duplicates(subset="code", keep="first")
    
    

['Coca Cola 50Cl',
 'Coca Cola 500ml',
 'Coca-Cola 50 Cl',
 'CocaCola 50Cl',
 'Coca-Cola 500ml',
 'Coca Cola .50L',
 'Coca Cola0,5L',
 'Coca Cola.5L',
 ' cocacola 50cl',
 'coke cola 50cl']

In [86]:
converting_table_1_litre = {
    "l": 1,
    "cl": 100,
    "dl": 10,
    "ml": 1000
}

converting_table_1_gram = {
    "mg": 1000,
    "kg": 0.1
}


In [92]:
def detect_unit(product_name: str):
    pass

def convert_unit(product_name: str, subunit: str, conversion_table: dict):
    pass


'0000000003087'